cartographer_grpc/framework/execution_context.h

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2017 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CARTOGRAPHER_GRPC_FRAMEWORK_EXECUTION_CONTEXT_H
+#define CARTOGRAPHER_GRPC_FRAMEWORK_EXECUTION_CONTEXT_H
+
+#include "cartographer/common/make_unique.h"
+#include "cartographer/common/mutex.h"
+#include "glog/logging.h"
+
+namespace cartographer_grpc {
+namespace framework {
+
+// Implementations of this class allow RPC handlers to share state among one
+// another. Using Server::SetExecutionContext(...) a server-wide
+// 'ExecutionContext' can be specified. This 'ExecutionContext' can be retrieved
+// by all implementations of 'RpcHandler' by calling
+// 'RpcHandler::GetContext<MyContext>()'.
+class ExecutionContext {
+ public:
+  // This non-movable, non-copyable class is used to broker access from various
+  // RPC handlers to the shared 'ExecutionContext'. Handles automatically lock
+  // the context they point to.
+  template <typename T>
+  class Handle {
```

Comment:
The name "Handle" reminds me more of on ID. What it does is providing synchronized/synchronous access. How about the name "Synchronized" or "LockedAccess" or "SharedAccess"?

Reply:
I like "Synchronized<Synchronized>"